### ISSUE TITLE
Our filter does not work properly with FilteredClassifer

### ISSUE THREAD

**Issue by StephenNu** [StephenNu/HDA#64]
It was reported that

> The filter runs successfully in the Preprocess panel but does not produce good results in conjunction with the FilteredClassifier (i.e., performance in a cross-validation is very poor). When the filter is fed multiple batches of data (e.g., a training set and a test set), the second (and subsequent) batches of data need to be transformed based on the filtering model built from the first batch of data.
# Todo
- figure out what needs to be changed so future batches of data are transformed, so the classifier is properly trained.
